### PR TITLE
add documentation for mock tooling

### LINKED ISSUE #177: Document mock tooling
Documentation should go to an extra file, align with other 2.0.0 documentation issues.

Markdown and doctype comments should be in sync.

### PR DESCRIPTION
**Pull request by ISO50**
close #177

### CODE REVIEW

**Comment by ISO50** [comment]
👍

File: API.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+It helps to easily write integration tests without writing extensive boilerplate code and without disabling oAuth for tests.
+
+This tooling is based on Nock, a HTTP mocking library. For more information about Nock see [Nock documentation](https://github.com/node-nock/nock).
+
+### mockAccessTokenEndpoint
+
+Creates a __very basic__ mock of token endpoint as defined in [RFC 6749](https://tools.ietf.org/html/rfc6749).
+
+The mocked endpoint will return a token(Link to type) with the scopes specified in the request.
```
Comment (Retro64) [inline API.md]: Link to Type seems to be a stub

File: API.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+* ⚠️ The mock does not validate the request
+* ⚠️ The mock holds a state that contains the created tokens
+* ⚠️ `cleanMock` resets the state and removes all nocks
```
Comment (Retro64) [inline API.md]: maybe emphasize *all*

File: API.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  * `times?: number` - defines number of calls the endpoint is mocked, default is `Number.MAX_SAFE_INTEGER`
+
+### mockTokenInfoEndpoint
+
+Creates a __very basic__ mock of a token validation endpoint.
+
+Returns `200` and a Token object if the given token is valid (via query parameter).
+The token is valid:
+
+* If it's specified via options parameter
```
Comment (Retro64) [inline API.md]: it is looks less casual

File: API.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+#### Arguments
+
+* `options`:
+  * `url: string` - url of the Token validation endpoint
+  * `times?: number` - defines number of calls the endpoint is mocked, default is `Number.MAX_SAFE_INTEGER`
+* `tokens: Token[]` -  optional list of valid tokens and their scopes.
+
+### cleanMock()
+
+* Remove all `nock` mocks (not only from this lib, really ALL)
```
Comment (Retro64) [inline API.md]: maybe emphasize *all*

File: API.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
+# Authmosphere API documentation
+
+## Mock tooling
+
+This tooling provides an abstraction to easily mock oAuth 2.0 (RFC) related endpoints.
```
Comment (bzums) [inline API.md]: We should use `OAuth` instead `oAuth` consistently I think.

Comment (bzums) [reply API.md]: RFC should be a link I guess?

File: API.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+#### Signature
+
+`mockAccessTokenEndpoint(options) => nock.Scope`
+
+#### Arguments
+
+* `options`:
+  * `url: string` - url of the Token validation endpoint
+  * `times?: number` - defines number of calls the endpoint is mocked, default is `Number.MAX_SAFE_INTEGER`
+
```
Comment (bzums) [inline API.md]: I think a horizontal line would help to separate the methods (it is difficult to see where one doc ends and the next begins currently)

File: API.md (hunk 1 above)
Comment (ISO50) [reply API.md]: ✅

File: API.md (hunk 6 above)
Comment (bzums) [reply API.md]: Or is there maybe some other way to emphasize the separation? Can also think about that later

File: API.md (hunk 4 above)
Comment (ISO50) [reply API.md]: ✅

File: API.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+});
+```
+
+#### Signature
+
+`mockAccessTokenEndpoint(options) => nock.Scope`
+
+#### Arguments
+
+* `options`:
```
Comment (bzums) [inline API.md]: we have a type for `options`, right? shall we name it here and link it maybe?

Comment (bzums) [reply API.md]: Nevertheless I think it is good to mention the properties explicitly here!

File: API.md (hunk 2 above)
Comment (ISO50) [reply API.md]: See clean mock part.

File: API.md (hunk 5 above)
Comment (ISO50) [reply API.md]: ✅

File: API.md (hunk 7 above)
Comment (ISO50) [reply API.md]: Added a link for all `options` ;)

File: API.md (hunk 6 above)
Comment (ISO50) [reply API.md]: I thought about using HR as seperator between different parts of the documentation

**Comment by bzums** [comment]
👍

**Comment by ISO50** [comment]
👍

**Comment by ISO50** [comment]
👍

**Comment by bzums** [comment]
👍

**Pull request merged**